### DISCUSSION TITLE
Are there any examples on ways organizations manage outsorced system development to follow organization's open source policies and processes?

### DISCUSSION THREAD

**Discussion by anajsana** [todogroup/ospology#134] [OSPO Discussions]
Question raised via [TODO Slack](https://thetodogroup.slack.com/archives/C0GBYUWCB/p1655185669432059)

Do you know if there are any existing case studies focused on "system integrators"?. In Japan, "system integration" is a business model referred to outsourced system development. In such a business entity, there are some restrictions to sharing codebases or artifacts based on customer requirements EVEN INTERNALLY.  As a result, this situation could make it difficult to define/create/manage a clear process.

**Comment by anajsana** [comment]
@anajsana answered:

*By System integration, do you mean ways outsourced development from contractors follow the organization's OS policies and processes?*

**Reply by anajsana** [reply]
Taniguchi answered:

The contractor you mention above would be the System integrator, I think. It could be said offshoring development, but I'm not exactly sure. The points are like this:

* Company A develops/builds  whole/a part of Customer (Company B) 's system
* "A" is responsible to develop/maintain the system based on "B"'s requirements.
* Basically, all the artifacts such as code base belong to the "B"

Here, company "A" is the System Integrator or "SIer" we call. Because of the 3rd point, open sourcing or inner-sourcing becomes difficult for Company "A".

**Reply by anajsana** [reply]
Mats Wichmann replied:

> Because of the 3rd point, open sourcing or inner-sourcing becomes difficult for Company "A".

*It seems like this would be down to what the contract terms were between A and B, and if after the fact, anything can be renegotiated.*

**Reply by annania** [reply]
I agree with Mats response. Employees often need approval to develop open source on Company time and with Company resources (equipment) and in doing so, the Company may own the copyrights. Similarly, if Company A is developing on behalf of Company B, then the contract may read that all work products are owned by Company B. Company B would then need to approve the contributions. In my experience, I was a developer on a contract that read like this and I sent a letter to "Company B" explaining the situation and asked for permission which they granted.
